cypress/integration/checkin/checkin.basic.spec.js

```diff
@@ -0,0 +1,127 @@
+    cy.getAdminToken().then(() => {
+      cy.getInstanceTypes({ limit: 1 }).then((instanceTypes) => { itemData.instanceTypeId = instanceTypes[0].id; console.log('itemData.instanceTypeId= ' + itemData.instanceTypeId); });
+      cy.getHoldingTypes({ limit: 1 }).then((res) => { itemData.holdingTypeId = res[0].id; });
+      cy.getLocations({ limit: 1 }).then((res) => { itemData.locationId = res.id; });
+      cy.getLoanTypes({ limit: 1 }).then((res) => { itemData.loanTypeId = res[0].id; });
+      cy.getMaterialTypes({ limit: 1 }).then((res) => { itemData.materialTypeId = res.id; });
+      ServicePoints.getViaApi({ limit: 1 }).then((servicePoints) => { itemData.servicepointId = servicePoints[0].id; });
+    }).then(() => {
+      InventoryInstances.createFolioInstanceViaApi({ instance: {
+        instanceTypeId: itemData.instanceTypeId,
+        title: itemData.instanceTitle,
+      },
+      holdings: [{
+        holdingsTypeId: itemData.holdingTypeId,
+        permanentLocationId: itemData.locationId,
+      }],
+      items:[{
+        barcode: itemData.barcode,
+        status:  { name: 'Available' },
+        permanentLoanType: { id: itemData.loanTypeId },
+        materialType: { id: itemData.materialTypeId },
+      }] });
+    }).then(specialInstanceIds => {
+      itemData.testInstanceIds = specialInstanceIds;
+    });
+
+    cy.createTempUser(userData.permissions)
+      .then(userProperties => {
+        userData.username = userProperties.username;
+        userData.password = userProperties.password;
+        userData.userId = userProperties.userId;
+        userData.barcode = userProperties.barcode;
+        userData.firstName = userProperties.firstName;
+      })
+      .then(() => {
+        UserEdit.addServicePointViaApi(itemData.servicepointId,
+          userData.userId, itemData.servicepointId);
+
+        const payloadInCheckOutItemRequest = {
+          id: uuid(),
+          itemBarcode: itemData.barcode,
+          loanDate: moment.utc().format(),
+          servicePointId: itemData.servicepointId,
+          userBarcode: userData.barcode,
+        };
+
+        checkoutActions.createItemCheckoutApi(payloadInCheckOutItemRequest);
+
+        cy.login(userData.username, userData.password, { path: TopMenu.checkInPath, waiter: CheckInActions.waitLoading });
+      });
+  });
+
+  after('Delete New Service point, Item and User', () => {
+    InventoryInstances.deleteInstanceViaApi(itemData.barcode);
+    cy.get('@userProperties').then(userProperties => {
+      Users.deleteViaApi(userProperties.userId);
+    });
+  });
+
+  it('C347631 Check in: Basic check in', { tags: [TestTypes.smoke, TestTypes.broken] }, () => {
+    CheckInActions.checkInItemGui(itemData.barcode);
+    InTransitModal.verifyModalTitle();
+    InTransitModal.verifySelectedCheckbox();
+    InTransitModal.unselectChechbox();
+    InTransitModal.verifyUnSelectedCheckbox();
+    InTransitModal.closeModal();
+    cy.get('@userProperties').then(userProperties => {
```

Comment:
don't use cy env

Reply:
@dmitriyfolio  Can you advise another solution of sending the values in async cypress functions? 
we can share the context between files using the wrap function in cypress. It is approved to do this way in cypress. Also  there is way of sharing the context using cy.task. When we use a synchronized way we can use variables defined in describe section or in fixtures. 
What do you advise to do in this way?

Reply:
try write userData instead of cy.get('@userProperties').then(userProperties
For example:
`cy.get('@userProperties').then(userProperties => {
      CheckInActions.checkActionsMenuOptions();
      cy.do(loanDetailsButton.click());
      cy.expect(Pane(including(userProperties.username)).exists());
      cy.expect(Pane(including('Loan details')).exists());
      returnCheckIn();
      cy.do(patronDetailsButton.click());
      cy.expect(Pane({ title: including(userProperties.username) }).exists());
      returnCheckIn();
      cy.do(itemDetailsButton.click());
      cy.expect(Pane(including(itemData.barcode)).exists());
      returnCheckIn();
      cy.do(newFeeFineButton.click());
      cy.expect(Modal(including('New fee/fine')).exists());
      returnCheckIn();
    });`
to: 
`      CheckInActions.checkActionsMenuOptions();
      cy.do(loanDetailsButton.click());
      cy.expect(Pane(including(userData.username)).exists());
      cy.expect(Pane(including('Loan details')).exists());
      returnCheckIn();
      cy.do(patronDetailsButton.click());
      cy.expect(Pane({ title: including(userData.username) }).exists());
      returnCheckIn();
      cy.do(itemDetailsButton.click());
      cy.expect(Pane(including(itemData.barcode)).exists());
      returnCheckIn();
      cy.do(newFeeFineButton.click());
      cy.expect(Modal(including('New fee/fine')).exists());
      returnCheckIn();`